lib/features/snack_queue/domain/entities/top_snack_bar.dart

```diff
@@ -0,0 +1,26 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_template/features/snack_queue/domain/entities/snack_message_type.dart';
+
+/// Display information for the snack.
+@immutable
+class TopSnackBar {
+  /// Message in the snack.
+  final String message;
+
+  /// Type of message.
+  final SnackMessageType messageType;
+
+  /// Snack display time.
+  final DateTime showTime;
+
+  /// Create an instance [TopSnackBar].
+  TopSnackBar({
+    required this.message,
+    required this.messageType,
+  }) : showTime = DateTime.now();
```

Comment:
Why can't we control `showTime` and it is always assigned to `DateTime.now()`?

Reply:
This variable is used to store the time when the snack was shown. I don't understand why we have to pass it through the constructor.